frontend/pages/create-a-form.js

```diff
@@ -4,27 +4,213 @@ import Hero from '../components/Hero'
 import DefaultLayout from '../layouts'
 import Button from '../components/atoms/Button'
 import FormCard from '../components/atoms/FormCard'
-import { useContext, useEffect, useReducer } from 'react'
+import { useContext, useEffect, useReducer, useState } from 'react'
 import SVG from 'react-inlinesvg'
 import UserContext from '../components/context/UserContext'
 import { Tabs, TabList, Tab, TabPanels, TabPanel } from "@reach/tabs";
 import classNames from 'classnames'
 import InputFile from '../components/atoms/InputFile'
+import Select from '../components/atoms/Select'
 import Unauthorized from '../components/pages/Unauthorized'
 import { useTranslations } from 'next-intl'
+import toast, { Toaster } from 'react-hot-toast'
+import Switch from 'react-switch'
+import Wysiwyg from '../components/molecules/Wysiwyg'
+import {
+  Listbox,
+  ListboxOption,
+} from "@reach/listbox";
+import api from '../libs/api'
+import {
+  MdAddCircleOutline,
+  MdClose,
+  MdDeleteOutline,
+  MdRadioButtonChecked,
+  MdCheckBox,
+  MdShortText,
+  MdContentCopy,
+  MdMoreVert
+} from 'react-icons/md'
+
+const COMPONENTS_TYPES = {
+  'Choice': CreateChoice,
+  'Checkbox': CreateCheckbox,
+  'Freeform': CreateFreeform,
+}
+
+const QUESTIONS_TYPES = [
+  'Checkbox',
+  'Freeform',
+  'Choice'
+]
+
+const QUESTIONS_ICONS = {
+  'Checkbox': MdCheckBox,
+  'Freeform': MdShortText,
+  'Choice': MdRadioButtonChecked,
+}
+
+const ACCESSIBILITY_OPTIONS = [
+  'Public',
+  'Link',
+  'FriendsOnly',
+  'Whitelist',
+]
 
 const types = {
+  SET_TITLE: "SET_TITLE",
+  SET_DESCRIPTION: "SET_DESCRIPTION",
+  SET_EXCERPT: "SET_EXCERPT",
+  SET_ACCESSIBILITY: "SET_ACCESSIBILITY",
+  SET_LIMITATIONS: "SET_LIMITATIONS",
+  ADD_QUESTION: "ADD_QUESTION",
+  DUPLICATE_QUESTION: "DUPLICATE_QUESTION",
+  REMOVE_QUESTION: "REMOVE_QUESTION",
+  ADD_QUESTION_INFO: "ADD_QUESTION_INFO",
+  SET_QUESTION_TYPE: "SET_QUESTION_TYPE",
+  REMOVE_QUESTION_INFO: "REMOVE_QUESTION_INFO",
+  EDIT_QUESTION_TITLE: "EDIT_QUESTION_TITLE",
+  EDIT_QUESTION_OPTIONAL: "EDIT_QUESTION_OPTIONAL",
+  EDIT_QUESTION_INFO: "EDIT_QUESTION_INFO",
   SET_ICON: "SET_ICON",
   SET_BANNER: "SET_BANNER",
 }
 
 const initialState = {
+  title: '',
+  description: '',
+  excerpt: '',
+  accessibility: 'Public',
+  limitations: null,
+  questions: [],
   icon: [],
   banner: [],
 }
 
 function reducer(state, action) {
```

Comment:
This function doing too much. You can move it to a map
```js
const apply = {
   types.SET_TITLE: (state, value) => state.title = value,
}
```
and change reducer to 
```js
return apply[action.type](...state, action.value)
```

Reply:
This is the recommended way in the React docs

Reply:
https://beta.reactjs.org/apis/usereducer#adding-a-reducer-to-a-component

Reply:
It is, but the reducer is too complex. Splitting things should not break anything I believe

Reply:
I don't see the point, here everything is in the same place it is easier to understand instead of splitting every case into functions. If I were to refactor this I would create a dedicated store if it becomes too complex

Reply:
(Thing that I will probably do but not now tbh @JustRoxy)

Reply:
(Just want to see if it works 😫)